state/state.go

```diff
@@ -371,8 +371,6 @@ func (s *State) UpgradeChannel(b *ui.Task, pkg *manifest.Package) (*manifest.Pac
 
 	if err != nil {
 		b.Warnf("Could not check updates for %s. Skipping update. Error: %s", name, err)
-	} else if etag == "" {
```

Comment:
I am not quite sure how this fixes the issue with empty DB. `etag` here is the one returned from the http request, I think it would be empty only of the source does not return the etag header.
With this change, we would still not upgrade (as presumable the etag in the DB would be an empty string as well), but we would just not get a warning. Not sure if that is desired.

Reply:
To clarify, that warning is there to let the user know that the source does not return etags. We decide not to download and reinstall the artifact in that case, but log a warning instead. I suppose that warning is not very descriptive, though.

Reply:
Mmm yeah. Will push another change.

Reply:
So I think I now know what was happening.

ETag checks use the fail-fast HTTP client. Because I was in a poor Internet area this was timing out on every execution of the binary. I think in this situation we want to update the `UpdatedAt` field regardless of whether we get a result, so at least it doesn't keep re-checking.

Secondarily there is currently a feature where if the DB is deleted, archives will be downloaded twice. I believe this is because `evictPackages()` nukes the cached archive because it sees an empty DB ETag differing from reality.

Reply:
That makes sense. And yes, updating UpdatedAt field seems reasonable if the connection fails 👍